h2o-core/src/test/java/water/TestUtil.java

```diff
@@ -450,7 +452,35 @@ public static Frame parse_test_file( Key outputKey, String fname, ParseSetupTran
     return ParseDataset.parse(outputKey, new Key[]{nfs._key}, true, guessedSetup);
   }
 
-  protected Frame parse_test_file( String fname, String na_string, int check_header, byte[] column_types) {
+  private static final String TEST_DATA_PATH_PREFIX_KEY = "H2O_TEST_DATA_PATH_PREFIX";
+
+  /**
+   * @param fileName File to prefix with local test file path
+   * @return Original path prefixed with local small data path from environment variable, if the environment variable is
+   * present. Otherwise the original path.
+   */
+  private static String localTestDataPath(final String fileName) {
+    Objects.requireNonNull(fileName);
+
+    final String smallDataPathPrefix = System.getenv(TEST_DATA_PATH_PREFIX_KEY);
+    if (smallDataPathPrefix == null) return fileName;
+
+    final StringBuilder localizedFileNameBuilder = new StringBuilder(smallDataPathPrefix);
+
+    if (!smallDataPathPrefix.endsWith("/")) {
+      localizedFileNameBuilder.append('/');
+    }
+
+    if (fileName.startsWith("./")) {
```

Comment:
this will work only if the property is set, is that intentional?

Reply:
Yes, no modification done unless it's not set.